### DISCUSSION TITLE
RAG difference in behavior when attaching image files vs text files

### DISCUSSION THREAD

**Discussion by mkagit** [danny-avila/LibreChat#4165] [Q&A]
I was running some tests here with RAG and noticed there's a difference when you upload and attach a text file, it could be PDF, Markdown, or Docx, compared to when you upload and attach an image file. When you attach an image file and have multiple models, at least in my case, where I'm using OpenRouter, while changing the models in the same chat, it keeps using, for example, in my case here, GPT-4o-mini for all models chosen. Is this behavior expected?

**Comment by danny-avila** [comment]
Yes, it's using the first found vision model. Some way to customize this is in order, but the current behavior is expected. Uploading images has nothing to do with RAG since it goes to AI provider API directly.

**Reply by mkagit** [reply]
I see, thanks. But even after the vision model first reply if I try to continue the chat with another model even when using the same custom endpoint but using another model it's keeping to use the first vision model.

**Reply by mkagit** [reply]
Still, in the file manager you can't see what is a RAG file or what is a regular image that goes to the provider API, as for all files are set as "host" in the storage field.
![Captura de tela_23-9-2024_95923_gpete com br](https://github.com/user-attachments/assets/e5e33a40-4831-49b7-b895-864fa4bb7d88)
Btw, some thumbnails failing to show in the file manager.
